burn-autodiff/src/ops/tensor.rs

```diff
@@ -1077,6 +1104,34 @@ impl<B: Backend> TensorOps<Self> for Autodiff<B> {
         }
     }
 
+    fn prod_dim<const D: usize>(tensor: FloatTensor<Self, D>, dim: usize) -> FloatTensor<Self, D> {
+        #[derive(Debug)]
+        struct ProdDim;
+
+        impl<B: Backend, const D: usize> Backward<B, D, 1> for ProdDim {
+            type State = (Shape<D>, usize);
+
+            fn backward(self, ops: Ops<Self::State, 1>, grads: &mut Gradients) {
+                let (shape, dim) = ops.state;
+
+                unary::<B, D, D, _>(ops.parents, ops.node, grads, |grad| {
+                    let ones = B::ones(shape, &B::device(&grad));
+                    let grad = B::prod_dim(grad, dim);
```

Comment:
As an example, I think the derivative of prod_dim([a, b, c, d], 0) should be 
[bcd, acd, abd, bcd], then multiply by the unchanged grad. So you'll need to register the original input as state. Don't hesitate to call me out if you think I'm wrong I haven't thought of it too long

Reply:
yeah i am stupid. i will add unit tests for this.